src/rest/domain/Stream.js

```diff
@@ -47,16 +47,25 @@ export default class Stream {
 
     async hasPermission(operation, userId) {
         const permissions = await this.getPermissions()
-        return permissions.find((p) => p.operation === operation && ((userId == null && p.anonymous) || (userId != null && p.user === userId)))
+        // eth addresses may be in checksumcase, but userId from server has no case
+        const userIdCaseInsensitive = typeof userId === 'string' ? userId.toLowerCase() : userId
+        return permissions.find((p) => (
```

Comment:
why not smth like this:
```
res = undefined // or what is expected if not found?
if ((userId == null && p.anonymous) || (userId != null && p.user.toLowerCase() === userIdCaseInsensitive)) {
res = permissions.find((p) =>p.operation === operation) 
}
return res
```
now this construction is unreadable

Reply:
Agreed this could read better, though my change was simply reformatting what was already there to fit under the max line limit after adding `.toLowerCase` and `s/userId/userIdCaseInsensitive`

Reply:
@mirpo changed in https://github.com/streamr-dev/streamr-client-javascript/pull/126/commits/faba92fba0b2e5e55b93f93751639e55616abfbd